src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -39,18 +39,69 @@ public static bool IsReservedKeyword(SyntaxKind kind)
 
         public static bool IsControlKeyword(SyntaxKind kind)
         {
-            return kind == SyntaxKind.YieldKeyword
-                || kind == SyntaxKind.WhenKeyword
-                || kind == SyntaxKind.InKeyword
-                || (kind != SyntaxKind.LockKeyword 
-                    && kind >= SyntaxKind.IfKeyword 
-                    && kind <= SyntaxKind.ThrowKeyword);
+            switch (kind)
+            {
+                case SyntaxKind.IfKeyword:
+                case SyntaxKind.ElseKeyword:
+                case SyntaxKind.WhileKeyword:
+                case SyntaxKind.ForKeyword:
+                case SyntaxKind.ForEachKeyword:
+                case SyntaxKind.DoKeyword:
+                case SyntaxKind.SwitchKeyword:
+                case SyntaxKind.CaseKeyword:
+                case SyntaxKind.DefaultKeyword:
```

Comment:
how is 'default' a control keyword?  for example.  "int i = default".  that's not a control keyword.

Reply:
but you don't do that check *inside* this public method.  Either that check needs to be here, or this should not be a public function.  Because the name is truly "IsThisAFlowControlKeywordAsLongAsCallerChecksTheParentIsValid".

Reply:
I am moving the IsControl* functions into ClassificationHelper and marking private.

Reply:
That works for me.  Thanks! :)